core/prisma/migrations/20250402192249_add_form_id_to_all_memberships/migration.sql

```diff
@@ -0,0 +1,106 @@
+- A unique constraint covering the columns `[pubId,memberGroupId,formId]` on the table `pub_memberships` will be added. If there are existing duplicate values, this will fail.
+- A unique constraint covering the columns `[stageId,userId,formId]` on the table `stage_memberships` will be added. If there are existing duplicate values, this will fail.
+- A unique constraint covering the columns `[stageId,memberGroupId,formId]` on the table `stage_memberships` will be added. If there are existing duplicate values, this will fail.
+
+*/
+-- DropIndex
+DROP INDEX "community_memberships_communityId_memberGroupId_key";
+
+-- DropIndex
+DROP INDEX "community_memberships_communityId_userId_key";
+
+-- DropIndex
+DROP INDEX "pub_memberships_pubId_memberGroupId_key";
+
+-- DropIndex
+DROP INDEX "pub_memberships_pubId_userId_key";
+
+-- DropIndex
+DROP INDEX "stage_memberships_stageId_memberGroupId_key";
+
+-- DropIndex
+DROP INDEX "stage_memberships_stageId_userId_key";
+
+-- CreateIndex
+CREATE UNIQUE INDEX "community_memberships_communityId_userId_formId_key" ON "community_memberships" ("communityId", "userId", "formId")
+WHERE
+    "formId" IS NOT NULL;
+
+CREATE UNIQUE INDEX "community_memberships_communityId_userId_key" ON "community_memberships" ("communityId", "userId")
+WHERE
+    "formId" IS NULL;
+
+-- CreateIndex
+CREATE UNIQUE INDEX "community_memberships_communityId_memberGroupId_formId_key" ON "community_memberships" ("communityId", "memberGroupId", "formId")
+WHERE
+    "formId" IS NOT NULL;
+
+CREATE UNIQUE INDEX "community_memberships_communityId_memberGroupId_key" ON "community_memberships" ("communityId", "memberGroupId")
+WHERE
+    "formId" IS NULL;
+
+-- CreateIndex
+CREATE UNIQUE INDEX "pub_memberships_pubId_userId_formId_key" ON "pub_memberships" ("pubId", "userId", "formId")
+WHERE
+    "formId" IS NOT NULL;
+
+CREATE UNIQUE INDEX "pub_memberships_pubId_userId_key" ON "pub_memberships" ("pubId", "userId")
+WHERE
+    "formId" IS NULL;
+
+-- CreateIndex
+CREATE UNIQUE INDEX "pub_memberships_pubId_memberGroupId_formId_key" ON "pub_memberships" ("pubId", "memberGroupId", "formId")
+WHERE
+    "formId" IS NOT NULL;
+
+CREATE UNIQUE INDEX "pub_memberships_pubId_memberGroupId_key" ON "pub_memberships" ("pubId", "memberGroupId")
+WHERE
+    "formId" IS NULL;
+
+-- CreateIndex
+CREATE UNIQUE INDEX "stage_memberships_stageId_userId_formId_key" ON "stage_memberships" ("stageId", "userId", "formId")
+WHERE
+    "formId" IS NOT NULL;
+
+CREATE UNIQUE INDEX "stage_memberships_stageId_userId_key" ON "stage_memberships" ("stageId", "userId")
+WHERE
+    "formId" IS NULL;
+
+-- CreateIndex
+CREATE UNIQUE INDEX "stage_memberships_stageId_memberGroupId_formId_key" ON "stage_memberships" ("stageId", "memberGroupId", "formId")
+WHERE
+    "formId" IS NOT NULL;
+
+CREATE UNIQUE INDEX "stage_memberships_stageId_memberGroupId_key" ON "stage_memberships" ("stageId", "memberGroupId")
+WHERE
+    "formId" IS NULL;
```

Comment:
sorry, i forgot about this: could you also add constraints that only `role = contributor` can have `formId`?

Reply:
good call!